sql/src/main/java/io/crate/planner/Planner.java

```diff
@@ -88,20 +99,33 @@
 import java.util.Locale;
 import java.util.Map;
 import java.util.Set;
+import java.util.function.BooleanSupplier;
 
 @Singleton
 public class Planner extends AnalyzedStatementVisitor<PlannerContext, Plan> {
 
     private static final Logger LOGGER = Loggers.getLogger(Planner.class);
+    static String EXPIRED_LICENSE_ERROR_MESSAGE = "Statement is not allowed - License is expired";
 
     private final ClusterService clusterService;
     private final LogicalPlanner logicalPlanner;
     private final Functions functions;
+    private LicenseService licenseService;
```

Comment:
attributes here can be final

Reply:
it can't unless I duplicate the constructor code which I prefer not to do